src/main/kotlin/ee/carlrobert/codegpt/settings/service/custom/CustomServiceChatCompletionForm.kt

```diff
@@ -15,7 +15,10 @@ import javax.swing.JButton
 import javax.swing.JPanel
 import javax.swing.SwingUtilities
 
-class CustomServiceChatCompletionForm(state: CustomServiceChatCompletionSettingsState) {
+class CustomServiceChatCompletionForm(
+    state: CustomServiceChatCompletionSettingsState,
+    val getApiKey: () -> String?
```

Comment:
How about we pass the reference of `apiKeyField` instead and then just extract the key directly from the field inside the listener?

Reply:
We could, but then we would essentially just copy&paste the `getApiKey` into `CustomServiceChatCompletionForm`
https://github.com/carlrobertoh/CodeGPT/blob/4c2639d1bfad92ea7df3bd20079512b806071ddb/src/main/kotlin/ee/carlrobert/codegpt/settings/service/custom/CustomServiceForm.kt#L93

I didnt see any benefit to it, also in terms of capsulation I thought its cleaner if the `CustomServiceChatCompletionForm` does not have direct access to the `apiKeyField` Component.

Reply:
Sure, I'm fine either way. I usually forward the callbacks only when the consumer needs to handle the component's actions/events.